flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -710,6 +711,9 @@ public long count() throws Exception {
 	 * @see DataSet
 	 */
 	public UnsortedGrouping<T> groupBy(String... fields) {
+		if (Keys.checkForAtomicType(getType(), fields)) {
```

Comment:
Sorry, just saw that you are doing exactly that in `checkForAtomicType()`, very good! :-)